presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogFilter.java

```diff
@@ -0,0 +1,288 @@
+            ImmutableMap<Integer, Predicate<String>> numberedGroupFilters,
+            Integer maxEntries, boolean keepFirst)
+            throws FileNotFoundException
+    {
+        this.file = requireNonNull(file);
+        this.logPattern = requireNonNull(logPattern);
+        this.lineSeparator = requireNonNull(lineSeparator);
+        this.namedGroupFilters = requireNonNull(namedGroupFilters);
+        this.numberedGroupFilters = requireNonNull(numberedGroupFilters);
+        this.maxEntries = maxEntries; // null -> no maximum
+        this.keepFirst = keepFirst;
+        if (!Files.exists(file)) {
+            throw new FileNotFoundException(file.toString());
+        }
+    }
+
+    public Stream<String> streamEntries()
+            throws IOException, ConcurrentModificationException
+    {
+        LogEntries logEntryBuilders = Files.lines(file).collect(
+                LogEntries::new,
+                LogEntries::addLine,
+                (a, b) -> {
+                    throw new ConcurrentModificationException();
+                });
+        return logEntryBuilders.streamEntries().map(StringJoiner::toString);
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    /**
+     * A builder for creating {@link LogFilter LogFilters}.
+     *
+     * Builder instances can be safely reused.
+     */
+    public static class Builder
+    {
+        private Path file;
+        private Pattern logPattern;
+        private String lineSeparator = System.getProperty("line.separator");
+        private ImmutableMap.Builder<String, Predicate<String>> namedGroupFilters = ImmutableMap.builder();
+        private ImmutableMap.Builder<Integer, Predicate<String>> numberedGroupFilters = ImmutableMap.builder();
+        private Integer maxEntries;
+        private boolean keepFirst = false;
+
+        private Builder() {}
+
+        public LogFilter build()
+                throws FileNotFoundException
+        {
+            return new LogFilter(file, logPattern, lineSeparator,
+                    namedGroupFilters.build(),
+                    numberedGroupFilters.build(),
+                    maxEntries, keepFirst);
+        }
+
+        public Builder setFile(Path file)
+        {
+            requireNonNull(file);
+            this.file = file;
+            return this;
+        }
+
+        /**
+         * Use the given regular expression to match each line of a log file
+         */
+        public Builder setPattern(Pattern pattern)
+        {
+            requireNonNull(pattern);
+            this.logPattern = pattern;
+            return this;
+        }
+
+        /**
+         * Add a filter to apply to a named capturing group
+         * in the regular expression for log file lines
+         */
+        public Builder addGroupFilter(String groupName, Predicate<String> filter)
+        {
+            requireNonNull(groupName);
+            if (filter != null) {
```

Comment:
Because a non-present parameter will be represented by `null` in `LogsHandler`, and if I ignore `null` filters, then I don't have to do a null check when adding them.